### PR TITLE
Intel® TensorFlow notebook container

### PR DESCRIPTION
**Pull request by sharvil10**
## Description

This PR adds containers for Intel® optimized TensorFlow with Intel® GPU and CPU support. There are 3 containers that are added. 

   1. The Base Image: This contains necessary driver level-zero installations for GPU support
   2. Runtime Image: This image contains the pip packages for Intel Extension for TensorFlow and TensorFlow 2.14.
   3. Jupyter image: This image adds necessary jupyterhub packages on top of the runtime image

## How Has This Been Tested?
This was tested using makefile's deploy, tests and validate commands.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by harshad16** [comment]
Hello @sharvil10
thank you for these amazing inclusion
if this is a part of some initiative, can you share it with us
or
if this is a community inclusion, 
can you please open an issue, with the details on requirements and how user can benefits from these

**Comment by sharvil10** [comment]
> Hello @sharvil10 thank you for these amazing inclusion if this is a part of some initiative, can you share it with us or if this is a community inclusion, can you please open an issue, with the details on requirements and how user can benefits from these

Hi @harshad16 thank you. Yes, this is part of Intel's initiative to enable Intel® optimized AI frameworks on OpenShift. Could you please share what information do you need? @Xaenalt can also provide more information this.

File: intel/base/gpu/ubi9-python-3.9/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      io.openshift.build.source-location="https://github.com/opendatahub-io/notebooks/tree/main/intel/base/gpu/ubi9-python-3.9" \
+      io.openshift.build.image="quay.io/opendatahub/workbench-images:intel-base-gpu-ubi9-python-3.9"
+
+USER 0
+WORKDIR /opt/app-root/src
+
+RUN . /etc/os-release && \
+    dnf install -y 'dnf-command(config-manager)' && \
+    dnf config-manager --add-repo \
+        https://repositories.intel.com/gpu/rhel/${VERSION_ID}/lts/2350/unified/intel-gpu-${VERSION_ID}.repo  && \
```
Comment (jstourac) [inline intel/base/gpu/ubi9-python-3.9/Dockerfile:20]: Hm, where is the `VERSION_ID` defined? :thinking:

Comment (sharvil10) [reply intel/base/gpu/ubi9-python-3.9/Dockerfile:20]: `VERSION_ID` is defined in `/etc/os-release` script and is sourced on line 17.

File: intel/runtimes/tensorflow/ubi9-python-3.9/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,64 @@
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+LABEL name="odh-notebook-intel-runtime-pytorch-ubi9-python-3.9" \
+    summary="Runtime Intel® optimized pytorch notebook image for ODH notebooks" \
```
Comment (braultatgithub) [inline intel/runtimes/tensorflow/ubi9-python-3.9/Dockerfile:5]: shouldn't that say "...optimized Tensorflow notebook....?

Comment (sharvil10) [reply intel/runtimes/tensorflow/ubi9-python-3.9/Dockerfile]: Fixed. Thank you.

**Review by braultatgithub — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -192,6 +192,16 @@ codeserver-ubi9-python-3.9: base-ubi9-python-3.9
...
 	$(call image,$@,intel/base/gpu/ubi9-python-3.9,$<)
 
+# Build and push intel-runtime-tensorflow-ubi9-python-3.9 image to the registry
+.PHONY: intel-runtime-tensorflow-ubi9-python-3.9
+intel-runtime-tensorflow-ubi9-python-3.9: intel-base-gpu-ubi9-python-3.9
+	$(call image,$@,intel/runtimes/tensorflow/ubi9-python-3.9,$<)
+
+# Build and push jupyter-intel-tensorflow-ubi9-python-3.9 image to the registry
+.PHONY: jupyter-intel-tensorflow-ubi9-python-3.9
+jupyter-intel-tensorflow-ubi9-python-3.9: intel-runtime-tensorflow-ubi9-python-3.9
```
Comment (harshad16) [inline Makefile:202]: Noticed you are building the jupyter one with runtime as base?
any specific reason for this ?

**Review by harshad16 — CHANGES_REQUESTED** [review]
Rest all looks good
just one question.
even though we merge pytorch, the question stands for both.
@sharvil10

File: Makefile (hunk 3 above)
Comment (sharvil10) [reply Makefile:202]: Our initial design was to re-use the runtime and only install jupyter dependencies on top. We will update it to separate the runtime and jupyter in the next release as that will be optimal.

Comment (harshad16) [reply Makefile:202]: ack, perhaps
in release, we should make the jupyter one and runtime timee both build from base.
i noticed, the runtime has, openai inclusion,
if that is also required in jupyter one, lets get this in 
and work on the rest in next one

Comment (sharvil10) [reply Makefile:202]: Agreed, we can work on building on top of the base image in the next release. The oneAPI basekit is needed for both containers.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

:+1:

**Pull request merged**